### ISSUE TITLE
fails to detect C++17's inline variable definitions

### ISSUE THREAD

**Issue by orchistro** [universal-ctags/ctags#1942]
Before I start reporting, for your information:
https://stackoverflow.com/questions/38043442/how-do-inline-variables-work

The name of the parser: C++

The command line you used to run ctags:

```bash
$ ctags --options=NONE -o - foo.h
```

The content of input file:

```C++
class foo
{
    public:
        static inline int mVar1; // missed
        int mVar2;
};

inline int gVar; // missed
```

The tags output you are not satisfied with:

```
foo     foo.h   /^class foo$/;" c
mVar2   foo.h   /^        int mVar2;$/;"        m       class:foo       typeref:typename:int
```
As you can see, the tags output is missing the difinition of `mVar2` and `gVar`.

The tags output you expect:

```
foo     foo.h   /^class foo$/;" c
mVar1   foo.h   /^        int mVar1;$/;"        m       class:foo       typeref:typename:int
mVar2   foo.h   /^        int mVar2;$/;"        m       class:foo       typeref:typename:int
gVar   foo.h   /^int gVar;$/;"        v       typeref:typename:int
```

The version of ctags:

```
$ ctags --version
Universal Ctags 0.0.0(), Copyright (C) 2015 Universal Ctags Team
Universal Ctags is derived from Exuberant Ctags.
Exuberant Ctags 5.8, Copyright (C) 1996-2009 Darren Hiebert
  Compiled: Dec  5 2018, 13:53:59
  URL: https://ctags.io/
  Optional compiled features: +wildcards, +regex, +iconv, +option-directory
```

How do you get ctags binary:

Just cloned it from github and built it locally.

**Comment by masatake** [comment]
@orchistro, thank you for reporting this issue.
With the following patch, ctags can capture inline'ed variable.
However, I wonder how typeref field should be.
@pragmaware, could you look at this?

```
[yamato@master]~/var/ctags-github% git diff
git diff
diff --git a/parsers/cxx/cxx_parser.c b/parsers/cxx/cxx_parser.c
index bc0a1267..56ed60f6 100644
--- a/parsers/cxx/cxx_parser.c
+++ b/parsers/cxx/cxx_parser.c
@@ -1519,7 +1519,8 @@ check_function_signature:
 	if(
 		g_cxx.uKeywordState &
 		(
-			CXXParserKeywordStateSeenInline | CXXParserKeywordStateSeenExplicit |
+			/* From c++-17, inline can be used as a modifier for a variable.  */
+			CXXParserKeywordStateSeenExplicit |
 			CXXParserKeywordStateSeenOperator | CXXParserKeywordStateSeenVirtual
 		)
 	)
[yamato@master]~/var/ctags-github% cat foo.cc
cat foo.cc
inline int v;
[yamato@master]~/var/ctags-github% make
make
./misc/gen-repoinfo main/repoinfo.h
make  all-am
make[1]: Entering directory '/home/yamato/var/ctags-github'
./misc/gen-repoinfo main/repoinfo.h
  CCLD     ctags
make[1]: Leaving directory '/home/yamato/var/ctags-github'
[yamato@master]~/var/ctags-github% ./ctags --fields='*'  -o - ./foo.cc
./ctags --fields='*'  -o - ./foo.cc
v	./foo.cc	/^inline int v;$/;"	kind:variable	line:1	language:C++	typeref:typename:int	roles:def
```

**Comment by orchistro** [comment]
@masatake Thank you for the quick patch. It works.
As to typeref field, I think it should stay as it is :-)

**Comment by pragmaware** [comment]
Yep, will take a look soon.

**Comment by pragmaware** [comment]
Fixed via #1944
